onnx/checker.cc

```diff
@@ -485,6 +501,20 @@ void check_model(const ModelProto& model) {
   check_graph(model.graph(), ctx, lex_ctx);
 }
 
+void VerifyFunctionNode(
+    const NodeProto& node,
+    const FunctionProto& func,
+    const CheckerContext& ctx,
+    const LexicalScopeContext& lex_ctx) {
+  // Create a temporary graphproto to hold the expanded subgraph
+  GraphProto g;
+  g.set_name("func_" + func.name() + "_expanded_subgraph");
+  // To Generate unique internal tensor names
+  // while preserving node's input/output names
```

Comment:
Before expand to graph, we should check whether the node's inputs and outputs are the same as defined in the function?

Reply:
The function does not have any class defined with io type constraints. It is only a proto class. So here what I did is to expand the function into a sub-graph while preserving the tensor names in input and output list. Then the actual nodes takes the tensors (partially or wholly) will do the checking job